kartothek/io_components/read.py

```diff
@@ -65,12 +84,18 @@ def dispatch_metapartitions_from_factory(
         assert base_df is not None
         base_df = cast(pd.DataFrame, base_df)
 
-        # Group the resulting MetaParitions by partition keys
-        merged_partitions = base_df.groupby(dataset_factory.partition_keys)
-        merged_partitions = merged_partitions["__partition__"].unique()
-        for row, labels in merged_partitions.iteritems():
+        # Group the resulting MetaParitions by partition keys or a subset of those keys
+        merged_partitions = base_df.groupby(
+            by=list(dispatch_by), sort=False, as_index=False
+        )
+        for group_name, group in merged_partitions:
+            if not isinstance(group_name, tuple):
+                group_name = (group_name,)
             mps = []
-            for label in labels:
+            logical_conjunction = list(
+                zip(dispatch_by, ["=="] * len(dispatch_by), group_name)
```

Comment:
This looks fishy to me, perhaps I'm missing something, is `len(group_name) == len(dispatch_by)`?

Reply:
yes, because `merged_partitions` is a grouped by `dispatch_by`:
`merged_partitions = base_df.groupby(by=list(dispatch_by), sort=False, as_index=False)`
and `group_name contains` the names of the columns we have grouped by. 
You can check out `test_read_dataset_as_dataframes_dispatch_by_multi_col` where the case of multiple columns is tested and `len(dispatch_by) > 1`